ui/src/shared/reducers/helpers/fields.js

```diff
@@ -0,0 +1,79 @@
+import _ from 'lodash'
+
+// fieldWalk traverses fields rescursively into args mapping fn on every
+// field
+export const fieldWalk = (fields, fn, acc = []) =>
```

Comment:
I am not sure the complexity of this function is justified. It looks like we only need it once in `getFieldsDeep` to filter out fields of type `"function"`. Maybe a function that returns a flattened list of fields would be more reusable?

```
export const flatFields = ([head, ...tail]) => {
  if (!head) {
    return []
  }

  if (head.type === 'field') {
    return [head, ...flatFields(tail)]
  } else if (head.type === 'func') {
    return [...flatFields(head.args || []), ...flatFields(tail)]
  }
}
```